appservice/src/main/java/nu/yona/server/AppServiceApplication.java

```diff
@@ -83,60 +80,81 @@ public void addCorsMappings(CorsRegistry registry)
 	}
 
 	@Bean
-	@Qualifier("appleMobileConfigSigningCertificate")
-	public X509Certificate appleMobileConfigSigningCertificate()
+	@Qualifier("appleMobileConfigSigningKeyStore")
+	public KeyStore appleMobileConfigSigningKeyStore()
 	{
-		AppleMobileConfigProperties properties = yonaProperties.getAppleMobileConfig();
-		if (!properties.isSigningEnabled())
+		String fileName = yonaProperties.getAppleMobileConfig().getSigningKeyStoreFile();
+		logger.info("Loading Apple mobile config signing key store from {}", fileName);
+		try (InputStream inStream = new FileInputStream(fileName))
+		{
+			KeyStore keyStore = java.security.KeyStore.getInstance("PKCS12");
+			keyStore.load(inStream, yonaProperties.getAppleMobileConfig().getSigningKeyStorePassword().toCharArray());
+			assertKeyStoreContent(keyStore);
+			return keyStore;
+		}
+		catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException e)
 		{
-			logger.info("Apple mobile config signing certificate not loaded, as signing is disabled");
-			return null;
+			throw YonaException.unexpected(e);
 		}
-		return loadCertificateFromFile(properties.getSigningCertificateFile());
 	}
 
-	private X509Certificate loadCertificateFromFile(String signingCertificateFile)
+	private void assertKeyStoreContent(KeyStore keyStore)
```

Comment:
Technically, this is no assert, so I would prefer the name `verifyKeyStoreContent`.